File: antd-dart/example/05_graph.dart
import 'package:antd/antd.dart';

/// Demonstrates graph entry (DAG node) operations.
void main() async {
  final client = AntdClient();

  try {
    // Create a graph entry
    final result = await client.graphEntryPut(
      'your_secret_key_hex',
      [], // no parents (root node)
      'content_hash_hex',
      [
        GraphDescendant(publicKey: 'descendant_pk_hex', content: 'desc_content_hex'),
      ],
    );
    print('Graph entry created at ${result.address} (cost: ${result.cost} atto)');

    // Read the graph entry
    final entry = await client.graphEntryGet(result.address);
    print('Owner: ${entry.owner}');
    print('Parents: ${entry.parents}');
    print('Content: ${entry.content}');
    print('Descendants: ${entry.descendants.length}');

    // Check existence
    final exists = await client.graphEntryExists(result.address);
    print('Exists: $exists');

    // Estimate cost
    final cost = await client.graphEntryCost('your_public_key_hex');
    print('Estimated cost: $cost atto');
  } on AntdError catch (e) {
    print('Error: $e');
  } finally {
    client.close();
  }
}

File: antd-rust/examples/05-graph.rs
use antd_client::{Client, GraphDescendant, DEFAULT_BASE_URL};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::new(DEFAULT_BASE_URL);

    // Create a graph entry (DAG node)
    let result = client
        .graph_entry_put(
            "your_secret_key_hex",
            &[],
            "content_hash_hex",
            &[GraphDescendant {
                public_key: "descendant_pk_hex".to_string(),
                content: "descendant_content_hex".to_string(),
            }],
        )
        .await?;
    println!("Graph entry at: {} (cost: {} atto)", result.address, result.cost);

    // Read the entry back
    let entry = client.graph_entry_get(&result.address).await?;
    println!("Owner: {}", entry.owner);
    println!("Descendants: {}", entry.descendants.len());

    // Check existence
    let exists = client.graph_entry_exists(&result.address).await?;
    println!("Exists: {}", exists);

    // Estimate cost
    let cost = client.graph_entry_cost("your_public_key_hex").await?;
    println!("Estimated cost: {} atto", cost);

    Ok(())
}
